src/v/raft/consensus.cc

```diff
@@ -1368,7 +1368,7 @@ ss::future<> consensus::do_start() {
         auto u = co_await _op_lock.get_units();
 
         read_voted_for();
-        bool initial_state = is_initial_state();
+        bool initial_state = _log->is_new_log();
```

Comment:
wow this is weird haha. presumably there is a point at which either create or open occurs and we could track that, or push the initialization steps down. anyway, i definitely accept that this may not be easy to do and probably more appropriate to do in a different PR.

Reply:
Yea agreed it is a bit odd, but I opted to translate 1:1 for now, and leave open/create separation for another time, since I think would also involve some broad changes